packages/commons/src/auth/jwt.ts

```diff
@@ -70,31 +92,27 @@ export const verifyJwtToken = async (
         }
         logger.error(`Error getting public key`);
         return callback(jwksSigningKeyError());
-      })().catch((error) => callback(error));
+      })().catch(callback);
     };
 
     return new Promise((resolve, reject) => {
       jwt.verify(
         jwtToken,
         getSecret,
-        { audience: acceptedAudiences },
+        { audience: config.acceptedAudiences },
         (err, decoded) => {
           if (err || !decoded) {
             logger.warn(`Token verification failed: ${err}`);
-
-            const unverifiedDecoded = decodeJwtToken(jwtToken, logger);
-            const authData =
-              unverifiedDecoded && readAuthDataFromJwtToken(unverifiedDecoded);
-            const { userId, selfcareId } = getUserInfoFromAuthData(authData);
-            reject(tokenVerificationFailed(userId, selfcareId));
-          } else {
-            resolve({ decoded });
+            const { userId, selfcareId } = extractUserInfoForFailedToken();
+            return reject(tokenVerificationFailed(userId, selfcareId));
           }
+          return resolve({ decoded });
         }
       );
     });
   } catch (error) {
-    logger.error(`Error verifying JWT token: ${error}`);
-    return Promise.reject(error);
+    logger.error(`Error building JWKS clients: ${error}`);
```

Comment:
Are we sure that the only errors that can arise inside this try block are from JWKS clients' instantiation?
The previous catch log message was more generic

Reply:
> Are we sure that the only errors that can arise inside this try block are from JWKS clients' instantiation? The previous catch log message was more generic

I think so, since it's the only logic beside the `jwt.verify`, in which we already handle any other exception in the rejection of the promise

Reply:
I agree, no need to change this message since the `error` will contain the real cause

Reply:
Yeah I think so too, I would not change the message, let's keep it more generic - it's also more aligned with the exception you throw, which is "tokenVerificationFailed"
```suggestion
    logger.error(`Error verifying JWT token: ${error}`);
```